docs/src/catalyst_applications/ode_simulation_performance.md

```diff
@@ -0,0 +1,318 @@
+
+### [Building the Jacobian symbolically](@id ode_simulation_performance_symbolic_jacobian)
+By default, OrdinaryDiffEq computes the Jacobian using [*automatic differentiation*](https://en.wikipedia.org/wiki/Automatic_differentiation) (however, using [*finite differences*](https://en.wikipedia.org/wiki/Finite_difference) is [also possible](https://docs.sciml.ai/DiffEqDocs/stable/features/performance_overloads/)). Since Catalyst builds its ODEs symbolically, it is able to *compute an analytic Jacobian symbolically*. Typically, this is advantageous when some rows of the Jacobian are highly dense (that is, some system species interact with most other species of the system).
+
+To use this option, simply set `jac=true` when constructing an `ODEProblem`:
+```@example ode_simulation_performance_3
+using Catalyst, OrdinaryDiffEq
+
+brusselator = @reaction_network begin
+    A, ∅ --> X
+    1, 2X + Y --> 3X
+    B, X --> Y
+    1, X --> ∅
+end
+u0 = [:X => 1.0, :Y => 0.0]
+tspan = (0.0, 20.0)
+p = [:A => 10.0, :B => 40.0]
+
+oprob = ODEProblem(brusselator, u0, tspan, p; jac=true)
+nothing # hide
+```
+
+### Using a sparse Jacobian
+For a system with $n$ variables, the Jacobian will be an $n\times n$ matrix. This means that, as $n$ becomes large, the Jacobian can become *very* large, potentially causing a significant strain on memory. In these cases, most Jacobian entries are typically $0$. This means that a [*sparse*](https://en.wikipedia.org/wiki/Sparse_matrix) Jacobian (rather than a *dense* one, which is the default) can be advantageous. To designate sparse Jacobian usage, simply provide the `sparse=true` option when constructing an `ODEProblem`:
+```@example ode_simulation_performance_3
+oprob = ODEProblem(brusselator, u0, tspan, p; sparse=true)
+nothing # hide
+```
+
+### [Linear solver selection](@id ode_simulation_performance_symbolic_jacobian_linear_solver)
+When implicit solvers use e.g. the Newton-Raphson method to, in each simulation time step, solve a (typically non-linear) equation, they actually solve a linearised version of this equation. For this they use a linear solver, the choice of which can impact performance. To specify one, we use the `linsolve` option (given to the solver function, *not* the `solve` command). E.g. to use the `KLUFactorization` linear solver we run
+```@example ode_simulation_performance_3
+using DifferentialEquations
+solve(oprob, Rodas5P(linsolve = KLUFactorization()))
```

Comment:
This is the wrong solver for anything of this size and we should demonstrate this more correctly.

Reply:
We should recommend `using MKL` and trying UMFPACK

Reply:
You mean that Rodas5P is wrong for large problems (where we would do something like this) and should be changed, or that since preconditioners and stuff like this is not good for a 2-state brusselators we should demonstrate on a larger model?

Reply:
Didn't find any examples for MKL, so was unsure on how to implement it here.